File: dsa_leetcode_beginner/0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.java
import java.util.*;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int val) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

class Solution {
    public int kthSmallest(TreeNode root, int k) {
        List<Integer> ioList = new ArrayList<>();
        inOrderTraversal(root, ioList);
        return ioList.get(k - 1);
    }

    private void inOrderTraversal(TreeNode node, List<Integer> ioList) {
        if (node == null) {
            return;
        }

        inOrderTraversal(node.left, ioList);
        ioList.add(node.val);
        inOrderTraversal(node.right, ioList);
    }
}

File: dsa_leetcode_beginner/0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.js
class TreeNode {
    constructor(val, left = null, right = null) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    kthSmallest(root, k) {
        const ioList = [];
        this.inOrderTraversal(root, ioList);
        return ioList[k - 1];
    }

    inOrderTraversal(node, ioList) {
        if (node === null) {
            return;
        }

        this.inOrderTraversal(node.left, ioList);
        ioList.push(node.val);
        this.inOrderTraversal(node.right, ioList);
    }
}
